update.go

```diff
@@ -338,8 +339,22 @@ func (u *Update) FromStream(updateWith io.Reader) (err error, errRecover error)
 			return
 		}
 	case PATCHTYPE_NONE:
-		// no patch to apply, go on through (with bzip2 decoding).
-		newBytes, err = ioutil.ReadAll(bzip2.NewReader(updateWith))
+		// no patch to apply, go on through
+		var fileHeader []byte
+		bufBytes := bufio.NewReader(updateWith)
+		fileHeader, err = bufBytes.Peek(2)
+		if err != nil {
+			return
+		}
+
+		if bytes.Equal([]byte{0x42, 0x5a}, fileHeader) {
```

Comment:
It would be better if the content format is specified somewhere as a flag, instead of checking magic bytes here. It's possible that one flat file starts with the same byte sequence, though very unlikely. It is at least worth a comment about the potential issue.